opentelemetry-jaeger/Cargo.toml

```diff
@@ -54,7 +58,11 @@ optional = true
 
 [features]
 default = []
-collector_client = ["isahc", "http"]
+collector_client = ["http", "opentelemetry/http"]
+isahc_collector_client = ["isahc", "collector_client"]
```

Comment:
I don't think we actually need to support isahc? in the other exporters it's just part of the example for "bring your own" right?

Reply:
Yes. But if we remove it, it would cause a breaking change for existing user. I think might as well just added a default implementation for `isahc`. So ideally existing users will only have to change the feature from `collector_client` to `isahc_collector_client` and everything should work just fine.

Reply:
Hm none of the isahc types were in the public api right? shouldn't codebases that were using it via the `collector_client` feature still work the same way now? There was never a way of passing the client manually so it shouldn't break any apps as far as I can see.

That being said we could also keep it and maybe mark as deprecated? Just seems like one we may not want to be maintaining over time

Reply:
Yep. So after merging this PR. The `collector_client` feature will contains no default http client. Users will have to call `with_http_client` explicitly to provide one otherwise it will throw a `NoHttpClient` error. That could break all of the current codebase.

Could mark `isahc_collector_client` to be deprecate and remove it later.

Reply:
Oh right, the ol' no reasonable default here, sounds fine then